### ISSUE TITLE
"Content-Disposition: inline" MIME Header is not working as expected

### ISSUE THREAD

**Issue by yasemindeki** [curl/curl#11429]
### I did this
I have been using the CURL Mime API in order to send an e-mail with embedded image (to display automatically when the message is viewed). I have tested this feature by coding your sample in [the link](https://curl.se/libcurl/c/smtp-mime.html). The only difference from the sample code is 
using 
`curl_mime_filedata(part, "designer.png");`
`curl_mime_encoder(part, "base64");`
instead of 
`curl_mime_filedata(part, "smtp-mime.c");`

The result is success of sending an e-mail with the image attachment but not inline. I have used the option of CURLOPT_DEBUGFUNCTION to understand the problem. The out data has the inline disposition where the multipart/alternative body part is.  So I have replaced the following code lines after the call of the curl_mime_encoder.

slist = curl_slist_append(NULL, "Content-Disposition: inline");
curl_mime_headers(part, slist, 1);

At this time, MIME entity related to image has the header field "Content-Disposition: inline". But the image is displayed as attachment on email client app not as inline.

### I expected the following
The library libcurl concerns the RFC 2183 specification [the link](https://curl.se/rfc/). The section 2.1 says 

> "A bodypart should be marked `inline' if it is intended to be displayed automatically"

So I think the library libcurl allows to send e-mail with a image as inline with it's Mime API. 

### curl/libcurl version
8.0.1

[DEBUGFUNCTION CURLINFO_DATA_OUT output]
Content-Type: multipart/mixed: boundary=-----------------------860af512b8c623bf
Mime-Version: 1.0
To:toaddress
From:fromaddress
Subject:subject

-------------------------860af512b8c623bf
Content-Type: multipart/alternative; boundary=-----------------------6eb9e29e26744027

-------------------------6eb9e29e26744027

> alternative body part of plain data

-------------------------6eb9e29e26744027

> alternative body part of html data

-------------------------6eb9e29e26744027--

-------------------------860af512b8c623bf
Content-Type: image/png
Content-Transfer-Encoding: base64
Content-Disposition: inline

> base64-encoded image file data

### operating system
Windows 7 Professional SP1 x64

**Comment by bagder** [comment]
> At this time, MIME entity related to image has the header field "Content-Disposition: inline". But the image is displayed as attachment on email client app not as inline.

This puzzles me. Isn't that header exactly what you say is missing and yet adding it does not seem to fix your issue? I don't understand what curl should do (differently) here.

**Comment by yasemindeki** [comment]
Thank you very much for your valuable support. My English is not very good, maybe I couldn't express my problem well. 

It took time to identify the cause of the problem. I've learned three ways to embed an image  in email. One of them, Content-ID (CID) image embedding has been done successfully with libcurl. 

But only MIME header Content-Disposition:inline should have been sufficient as RFC document says (without Content-ID (CID) image embedding). A lot of web articles mention that there are known issues with displaying in web-based email clients. So I've tested other some web clients and Thunderbird desktop. Thunderbird just dispayed the image successfully when message has viewed.

Finally libcurl has not a bug related with inline images.  I'm sorry it took time to realise. The problem arises from different behaviors of email clients.

Best regards
